transport/http/inbound_example_test.go

```diff
@@ -83,3 +84,52 @@ func ExampleMux() {
 	}
 	// Output: hello from /health
 }
+
+func ExampleInterceptor() {
+	// import nethttp "net/http"
+
+	// We create an interceptor which executes an override http.Handler if
+	// the HTTP request is not a well-formed YARPC request.
+	override := func(overrideHandler nethttp.Handler) http.InboundOption {
+		return http.Interceptor(func(transportHandler nethttp.Handler) nethttp.Handler {
+			return nethttp.HandlerFunc(func(w nethttp.ResponseWriter, r *nethttp.Request) {
+				if r.Header.Get("RPC-Encoding") == "" {
+					overrideHandler.ServeHTTP(w, r)
+				} else {
+					transportHandler.ServeHTTP(w, r)
+				}
+			})
+		})
+	}
```

Comment:
I like this example! The current code is great if the user wants a reusable `override` option, but it's more complex than it needs to be to demonstrate use of `Intercept`. If we move the `handler` definition below up, we can strip off a few layered functions:

```
fallback := nethttp.HandlerFunc(func(w nethttp.ResponseWriter, r *nethttp.Request) {
  io.WriteString(w, "hello, world!")
})

intercept := func(transportHandler nethttp.Handler) nethttp.Handler {
  return nethttp.HandlerFunc(func(w nethttp.ResponseWriter, r *nethttp.Request) {  
    if r.Header.Get("RPC-Encoding") == "" {
      // Not a YARPC request, use fallback handler.
      fallback.ServeHTTP(w, r)
    } else {
      transportHandler.ServeHTTP(w, r)
    }
  })
}

transport := http.NewTransport()
inbound := transport.NewInbound(":8889", http.Interceptor(intercept))
```